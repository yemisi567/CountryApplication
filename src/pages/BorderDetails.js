import { useState, useEffect } from "react";
import Loader from "../components/Loader";
import CountryDetails from "../components/CountryDetails";
import { useParams } from "react-router-dom";
import { useDispatch } from "react-redux";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { getCountryBorders } from "../redux/actions/countries";

Enzyme.configure({ adapter: new Adapter() });

const BorderDetails = () => {
  const dispatch = useDispatch();
  const [lang, setLang] = useState([]);
  const [loading, setLoading] = useState(false);
  const [currency, setCurrency] = useState([]);
  const [nativeName, setNativeName] = useState([]);
  const [countryDetails, setCountryDetails] = useState([]);
  const { countryCode } = useParams();

  const getSpecificBorders = async () => {
    setLoading(true);
    try {
      const res = await dispatch(getCountryBorders(countryCode));
      setCountryDetails(res.payload.data[0]);
      setLang(Object.values(res.payload.data[0].languages));
      setCurrency(Object.values(res.payload.data[0].currencies));
      setNativeName(Object.values(res.payload.data[0].name.nativeName));
      setLoading(false);
    } catch (err) {
      console.log(err);
    }
  };

  useEffect(() => {
    getSpecificBorders();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div>
      {loading === true ? (
        <Loader />
      ) : (
        <div>
          <CountryDetails
            imgSrc={countryDetails?.flags?.png}
            imgAlt={countryDetails?.name?.common}
            countryName={countryDetails?.name?.common}
            nativeName={nativeName}
            population={countryDetails?.population}
            region={countryDetails?.region}
            subregion={countryDetails?.subregion}
            capital={countryDetails?.capital}
            tld={countryDetails?.tld}
            currency={currency}
            lang={lang}
            borders={countryDetails?.borders}
          />
        </div>
      )}
    </div>
  );
};

export default BorderDetails;
