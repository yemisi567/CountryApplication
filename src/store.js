import { configureStore } from "@reduxjs/toolkit";
import countriesReducer from "./redux/reducers/reducer";

export const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
