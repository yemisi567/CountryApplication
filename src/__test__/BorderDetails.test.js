import { shallow } from "enzyme";
import BorderDetails from "../pages/BorderDetails";
import { Provider } from "react-redux";
import store from "../store";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("BorderDetails", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <BorderDetails />
      </Provider>
    );
  });

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
