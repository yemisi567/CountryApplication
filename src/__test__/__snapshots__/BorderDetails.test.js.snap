// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderDetails should render properly 1`] = `ShallowWrapper {}`;
